opba-api-security/src/main/java/de/adorsys/opba/api/security/external/mapper/HttpRequestToDataToSignMapper.java

```diff
@@ -75,4 +78,47 @@ public ConfirmConsentDataToSign mapToConfirmConsent(HttpServletRequest request,
 
         return new ConfirmConsentDataToSign(UUID.fromString(xRequestId), instant, OperationType.valueOf(operationType));
     }
+
+    public PaymentInitiationDataToSign mapToPaymentInititation(HttpServletRequest request, Instant instant) {
+        String xRequestId = request.getHeader(HttpHeaders.X_REQUEST_ID);
+        String operationType = request.getHeader(HttpHeaders.X_OPERATION_TYPE);
+
+        return new PaymentInitiationDataToSign(
+                UUID.fromString(xRequestId),
+                instant,
+                OperationType.valueOf(operationType),
+                request.getHeader(HttpHeaders.BANK_ID),
+                request.getHeader(HttpHeaders.FINTECH_USER_ID),
+                request.getHeader(HttpHeaders.FINTECH_REDIRECT_URL_OK),
+                request.getHeader(HttpHeaders.FINTECH_REDIRECT_URL_NOK),
+                null,
+                null
+        );
+    }
+
+    public GetPaymentDataToSign mapToGetPayment(HttpServletRequest request, Instant instant) {
+        String xRequestId = request.getHeader(HttpHeaders.X_REQUEST_ID);
+        String operationType = request.getHeader(HttpHeaders.X_OPERATION_TYPE);
+
+        return new GetPaymentDataToSign(
```

Comment:
Builder pattern would be safer here - long ctor.

Reply:
Lombok builder was added